text/0003-metadata-api.md

```diff
@@ -0,0 +1,129 @@
+- Feature Name: metadata_api
+- Start Date: 2021-07-13
+- RFC PR: (leave this empty)
+- Hathor Issue: (leave this empty)
+- Author: Giovane Costa <gigio.coder@gmail.com>
+
+# Summary
+[summary]: #summary
+
+Metadata API will provide metadata information about Transactions, Tokens, Blocks and other things on Network.
+It will start with Transactions and Tokens
+
+# Motivation
+[motivation]: #motivation
+
+We need to have some data to be used on Public Explore and Wallets that can't be stored on the blockchain about
+Transactions, Tokens, Blocks and other things on Network. 
+So, we store somewhere else and retrieve them through MetadataAPI.
+This data can also be used by community as they please.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+Information will be stored in S3 in format `(token|transaction)/[hash].json` inside the metadata bucket.
+
+A request made to the API passing `hash` and `type` will retrieve the information or `404` if not found.
+
+Types can be: `token` and `transaction` for now.  
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+**Endpoint:**
+
+GET `/metadata/:type/:hash` return `200` | `404`
+
+- `type` - type of the data (`token` or `transaction`)
+- `hash` - hash of the token or transaction
+
+
+`metadata-handler` return the stored metadata in the following json format
+
+```
+{
+  "id": string,
+  "type": string,
+  "data": (TokenMeta|TransactionMeta)
+}
+```
+
+```
+TokenMeta {
+  id: string,
+  verified: boolean,
+  banned: boolean,
+  reason: boolean,
+  nft: boolean,
+  nft_media: {
+    type: enum(VIDEO|IMAGE|AUDIO),
+    file: string,
+    loop: boolean
+  }
+}
+
+TransactionMeta {
+  id: string,
+  context: string, // a message to be shown on transaction page
+  genesis: boolean
+}
+
+```
+
+Examples:
+
+** Token Metadata **
+```
+{
+  "id": "00003aa356c9493464c657873b115c5e8667adf58cceeb4b37a1cdae0ddc9536",
+  "data": {
+    "id": "00003aa356c9493464c657873b115c5e8667adf58cceeb4b37a1cdae0ddc9536",
+    "verified": false,
+    "banned": false,
+    "reason": null,
+    "nft": true,
+    "nft_media": {
+      "type": "IMAGE",
+      "file": "http://curtis-white.com/mean/officer.png",
+      "loop": false
```

Comment:
Is it required even for image files? Does it work only for videos or audios can play in loop as well?

Reply:
It's not required in s3 json files but will be always present on responses.

Reply:
I think the documentation must have the details for each field.

Reply:
Added the datails on fields structure explanation